compiler/rustc_lint/messages.ftl

```diff
@@ -496,8 +496,9 @@ lint_requested_level = requested on the command line with `{$level} {$lint_name}
 
 lint_span_use_eq_ctxt = use `.eq_ctxt()` instead of `.ctxt() == .ctxt()`
 
-lint_supertrait_as_deref_target = `{$t}` implements `Deref` with supertrait `{$target_principal}` as target
-    .label = target type is set here
+lint_supertrait_as_deref_target = this `Deref` implementation is covered by the implicit super-trait coercion
+    .help = consider removing this implementation or replacing it with a method instead
+    .label = target type is a super-trait of `{$t}`
```

Comment:
No other lint calls it a super-trait.

```suggestion
lint_supertrait_as_deref_target = this `Deref` implementation is covered by the implicit supertrait coercion
    .help = consider removing this implementation or replacing it with a method instead
    .label = target type is a supertrait of `{$t}`
```

Reply:
Also, why did you remove `${target_principal}` here?

Reply:
> Also, why did you remove `${target_principal}` here?

It simply was used anymore. `@lcnr` suggested diagnostic output didn't included it.

Reply:
Either way works for me. If this lands first then I will rework the message to talk in the past tense, not the future tense. 

I wouldn't let this PR be blocked by a lint tweak.